pdata/plog/pb.go

```diff
@@ -24,7 +24,11 @@ func NewProtoMarshaler() Marshaler {
 	return newPbMarshaler()
 }
 
-// TODO(#3842): Figure out how we want to represent/return *Sizers.
+// NewProtoSizer returns a Sizer. Calculates the size of a marshaled Logs.
+func NewProtoSizer() Sizer {
```

Comment:
Not sure if this is the best option, we may consider alternative as having an interface `MarshalerSizer` similar with [`WriterSeeker`](https://pkg.go.dev/io#WriteSeeker)?

Reply:
I actually like that better. Would you expect the `NewProtoMarshaler` to return a `MarshalerSizer` or still a `Marshaler`?

Reply:
NewProtoMarshaler will return the most descriptive interface that implements, so `MarshalerSizer`

Reply:
@bogdandrutu Updated to use a `MarshlerSizer` interface. Let me know if you think there needs to be additional tweaks to this implementation.